dbt_platform_helper/commands/database_helpers.py

```diff
@@ -1,124 +1,205 @@
 from dbt_platform_helper.utils.aws import Vpc
-from dbt_platform_helper.utils.aws import get_aws_session_or_abort
 from dbt_platform_helper.utils.aws import get_connection_string
 from dbt_platform_helper.utils.aws import get_vpc_info_by_name
+from dbt_platform_helper.utils.messages import abort_with_error
+from dbt_platform_helper.utils.validation import load_and_validate_platform_config
 
 
-def run_database_copy_task(
-    session: boto3.session.Session,
-    account_id: str,
-    app: str,
-    env: str,
-    database: str,
-    vpc_config: Vpc,
-    is_dump: bool,
-    db_connection_string: str,
-):
-    client = session.client("ecs")
-    action = "dump" if is_dump else "load"
-    response = client.run_task(
-        taskDefinition=f"arn:aws:ecs:eu-west-2:{account_id}:task-definition/{app}-{env}-{database}-{action}",
-        cluster=f"{app}-{env}",
-        capacityProviderStrategy=[
-            {"capacityProvider": "FARGATE", "weight": 1, "base": 0},
-        ],
-        networkConfiguration={
-            "awsvpcConfiguration": {
-                "subnets": vpc_config.subnets,
-                "securityGroups": vpc_config.security_groups,
-                "assignPublicIp": "DISABLED",
-            }
-        },
-        overrides={
-            "containerOverrides": [
-                {
-                    "name": f"{app}-{env}-{database}-{action}",
-                    "environment": [
-                        {"name": "DATA_COPY_OPERATION", "value": action.upper()},
-                        {"name": "DB_CONNECTION_STRING", "value": db_connection_string},
-                    ],
-                }
-            ]
-        },
-    )
-
-    return response.get("tasks", [{}])[0].get("taskArn")
+class MaintenancePageProvider:
+    def __init__(self):
+        self.offline = offline_command
+        self.online = online_command
 
 
 class DatabaseCopy:
     def __init__(
         self,
-        account_id,
-        app,
-        env,
-        database,
-        vpc_name,
-        get_session_fn=get_aws_session_or_abort,
-        run_database_copy_fn=run_database_copy_task,
-        vpc_config_fn=get_vpc_info_by_name,
-        db_connection_string_fn=get_connection_string,
-        input_fn=click.prompt,
-        echo_fn=click.secho,
+        app: str,
+        database: str,
+        auto_approve: bool = False,
+        load_application_fn: Callable[[str], Application] = load_application,
+        vpc_config_fn: Callable[[Session, str, str, str], Vpc] = get_vpc_info_by_name,
+        db_connection_string_fn: Callable[
+            [Session, str, str, str, Callable], str
+        ] = get_connection_string,
+        maintenance_page_provider: Callable[
+            [str, str, list[str], str, str], None
+        ] = MaintenancePageProvider(),
+        input_fn: Callable[[str], str] = click.prompt,
+        echo_fn: Callable[[str], str] = click.secho,
+        abort_fn: Callable[[str], None] = abort_with_error,
```

Comment:
I haven't really looked at this thoroughly, but this feels like a lot of things going into the constructor.

It makes me wonder whether the `DatabaseCopy` class is doing a bit much. Maybe we can separate out the copying work from the input/output/side effects?

Reply:
I think some of this stuff can be tidied up further. For example the input/echo/abort are all UI elements that can probably be moved into a single UserInterfaceProvider or something. Perhaps in a tidy up step after we get this command out for FFT to use.